newIDE/app/src/UI/Search/UseSearchStructuredItem.js

```diff
@@ -86,6 +89,12 @@ export const filterSearchResults = <
 
   const startTime = performance.now();
   const filteredSearchResults = searchResults
+    .filter(
+      ({ item }) =>
+        !itemCategory ||
+        item.category === itemCategory ||
+        (itemCategory === 'General' && !item.category)
```

Comment:
This is a bit hard to read because I can't tell the difference between `item.category` and `itemCategory`. I feel like the name of `itemCategory` should be a bit more specific: `chosenItemCategory`? `requestedItemCategory`?